components/Common/Search/States/WithSearchBox.tsx

```diff
@@ -0,0 +1,187 @@
+
+    return () => reset();
+  }, []);
+
+  useEffect(() => {
+    search(searchTerm);
+  }, [searchTerm, selectedFacet]);
+
+  const search = (term: string) => {
+    orama
+      .search({
+        term,
+        ...DEFAULT_ORAMA_QUERY_PARAMS,
+        ...filterBySection(),
+      })
+      .then(setSearchResults)
+      .catch(setSearchError);
+  };
+
+  const reset = () => {
+    setSearchTerm('');
+    setSearchResults(null);
+    setSelectedFacet(0);
+  };
+
+  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    router.push(`/en/search?q=${searchTerm}&section=${selectedFacetName}`);
+    onClose();
+  };
+
+  const changeFacet = (idx: number) => {
+    setSelectedFacet(idx);
+  };
+
+  const filterBySection = () => {
+    if (selectedFacet === 0) {
+      return {};
+    }
+
+    return {
+      where: {
+        siteSection: {
+          eq: selectedFacetName,
+        },
+      },
+    };
+  };
+
+  const facets = {
+    all: searchResults?.count ?? 0,
+    ...(searchResults?.facets?.siteSection?.values ?? {}),
+  };
+
+  const selectedFacetName = Object.keys(facets)[selectedFacet];
+
+  return (
+    <div className={styles.searchBoxModalContainer}>
+      <div className={styles.searchBoxModalPanel} ref={searchBoxRef}>
+        <div className={styles.searchBoxInnerPanel}>
+          <div className={styles.searchBoxInputContainer}>
+            <button
+              onClick={onClose}
+              className={styles.searchBoxBackIconContainer}
+            >
+              <ChevronLeftIcon className={styles.searchBoxBackIcon} />
+            </button>
+            <MagnifyingGlassIcon
+              className={styles.searchBoxMagnifyingGlassIcon}
+            />
+            <form onSubmit={onSubmit}>
+              <input
+                ref={searchInputRef}
+                type="search"
+                className={styles.searchBoxInput}
+                onChange={event => setSearchTerm(event.target.value)}
+                value={searchTerm}
+              />
+            </form>
+          </div>
+
+          <div className={styles.fulltextSearchSections}>
+            {Object.keys(facets).map((facetName, idx) => (
+              <button
+                key={facetName}
+                className={classNames(styles.fulltextSearchSection, {
+                  [styles.fulltextSearchSectionSelected]: selectedFacet === idx,
+                })}
+                onClick={() => changeFacet(idx)}
+              >
+                {facetName}
+                <span className={styles.fulltextSearchSectionCount}>
+                  (
+                  {facets[facetName as keyof typeof facets].toLocaleString(
+                    'en'
+                  )}
+                  )
+                </span>
+              </button>
+            ))}
+          </div>
+
+          <div className={styles.fulltextResultsContainer}>
+            {searchError ? <WithError /> : null}
+
+            {(searchTerm ? (
```

Comment:
If you use Bool (&& and ||) won't need a `?? null` and I assume it would be easier to read